### ISSUE TITLE
Implement a function to initialize the child model based on the selected crossover strategy.

### ISSUE THREAD

**Issue by csmangum** [Dooders/AgentFarm#612]
Provide a **single entry point** that, given a **crossover strategy** and two parent models (or checkpoints), **builds a child `nn.Module`** and fills its weights (and any quantization-related state) so the child is ready for inference or downstream training. This sits **above** the per-tensor crossover logic from [#611](https://github.com/Dooders/AgentFarm/issues/611): #611 defines *how* tensors combine; #612 defines *how* a full child is constructed and initialized from that choice.

## Motivation

- Callers (scripts, evolution loops, tests) should not duplicate: architecture construction, device placement, `load_state_dict` / quantized checkpoint loading, and strategy dispatch.
- “Child” must be **well-defined**: same class and hyperparameters as parents, valid after crossover for both **float** and **quantized** checkpoints where the project supports them.

## Relationship to other work

- **Depends on / pairs with**: [#611](https://github.com/Dooders/AgentFarm/issues/611) — crossover strategies and `crossover_*_state_dict`-style helpers.
- **Reuse**: Existing loading paths such as `load_quantized_checkpoint` in `farm/core/decision/training/quantize_ptq.py` for quantized parents; existing `StudentQNetwork` / `BaseQNetwork` construction patterns used by `scripts/quantize_distilled.py`, `scripts/qat_distilled.py`, and validation tooling.

## Proposed API (sketch)

Something in the spirit of:

```text
initialize_child_from_crossover(
    parent_a: nn.Module | Path | state_dict,
    parent_b: nn.Module | Path | state_dict,
    strategy: CrossoverStrategy | str,
    *,
    architecture: ChildArchitectureSpec | None,  # or infer from parent_a
    rng: torch.Generator | int | None,
    device: torch.device | str | None,
    **strategy_kwargs,  # e.g. alpha, layer_groups
) -> nn.Module
```

**Behavioral contract**

1. **Resolve parents** to a pair of in-memory models or aligned `state_dict`s (document whether both must already be quantized or mixed float+quant is allowed).
2. **Infer or require** `input_dim`, `output_dim`, `hidden_size` / `parent_hidden` (and any other ctor args) so the child class matches parents.
3. **Instantiate** a fresh child module (same type as parents).
4. **Run** the selected crossover (#611) to produce a child `state_dict` (or apply crossover in-place on a clone of a template state).
5. **`load_state_dict`** on the child (strict=True or documented mapping for renamed quant stubs).
6. Return child in a documented default mode (`eval()` vs `train()`, device).

Optional convenience: overload that accepts **only** crossed `state_dict` + architecture spec, for testing or when crossover was computed elsewhere.

## Strategy dispatch

The initializer should **not** embed crossover math; it should **delegate** to strategy-specific functions from #611, for example:

| Strategy | Delegation |
|----------|------------|
| Random parameter / per-tensor | `crossover_uniform_tensors(...)` |
| Layer-based | `crossover_by_layers(...)` |
| Weighted average | `crossover_weighted_average(..., alpha=...)` |

The initializer owns **orchestration and I/O**; strategies own **tensor rules**.

## Edge cases to specify

- **Strict key alignment**: If parents differ in `state_dict` keys, fail fast with a clear error (unless a future issue adds adapters).
- **Quantization metadata**: Child must receive **consistent** int8 weights + scales (or equivalent) per #611; initializer documents whether it always works from dequantized float parents internally.
- **Determinism**: `rng` / seed plumbed into stochastic strategies only; layer-based and fixed-`α` average may be deterministic without RNG.
- **Device**: Either build on CPU then `.to(device)`, or construct on target device; pick one and document.

## Acceptance criteria

- [ ] One public function (or small facade module) that implements the flow above and is documented in docstring (and briefly in decision/training docs or script help if exposed to CLI).
- [ ] Works end-to-end in tests: two toy parents → child → forward pass on a fixed batch without shape/dtype errors.
- [ ] Integrates with existing checkpoint loading where applicable (e.g. `load_quantized_checkpoint` for quantized paths).
- [ ] No duplication of crossover logic inside the initializer; calls into #611’s operators.

## References

- Issue: https://github.com/Dooders/AgentFarm/issues/612  
- Related: https://github.com/Dooders/AgentFarm/issues/611  
- Loading: `farm/core/decision/training/quantize_ptq.py` (`load_quantized_checkpoint`)
